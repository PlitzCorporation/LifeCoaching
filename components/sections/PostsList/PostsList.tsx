import React from 'react';

import { getAllBlogPosts } from '@/helpers/pullBlogPost';

import ImagePlacer from '@/components/ui/ImagePlacer';

import { PostProps } from '@/types/posts';
import SolidButton from '@/components/ui/SolidButton';

const PostsList = async () => {
	const allPosts: PostProps[] = await getAllBlogPosts();

	const sortedPosts = allPosts
		.filter((post) => post.isLive)
		.sort((a, b) => Date.parse(b.createdAt) - Date.parse(a.createdAt));

	return (
		<div className="plitz-container py-10 lg:py-20">
			{sortedPosts.length > 0 &&
				sortedPosts.map((post) => (
					<div
						key={post._id}
						className="grid grid-cols-1 lg:grid-cols-3 mb-20 last:mb-0 shadow-plitz-box-4-32"
					>
						<ImagePlacer imageSrc={post.othersizes.tablet} />
						<div className="relative lg:col-span-2">
							<div className="hidden lg:block h-[10px] bg-plitz-primary"></div>
							<div className="h-full flex flex-col justify-center px-8 lg:px-20 py-8 lg:py-10">
								<h3 className="text-balance mb-3 lg:mb-5">{post.title}</h3>
								<small className="text-plitz-primary mb-7 lg:mb-10">
									By {post.fullname} |{' '}
									{new Date(post.createdAt).toLocaleDateString('en-US', {
										month: 'long',
										day: '2-digit',
										year: 'numeric',
									})}
								</small>
								<div
									dangerouslySetInnerHTML={{ __html: post.postintro }}
									className="text-plitz-body mb-7 lg:mb-10"
								/>
								<div className="flex">
									<SolidButton
										label="Continue Reading"
										href={`/blog/${post.link}`}
									/>
								</div>
							</div>
						</div>
					</div>
				))}
		</div>
	);
};

export default PostsList;
