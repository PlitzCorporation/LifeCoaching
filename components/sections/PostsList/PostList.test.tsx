import PostsList from './PostsList';
import { render } from '@testing-library/react';

jest.mock('next/image', () => ({
	__esModule: true,
	// eslint-disable-next-line @typescript-eslint/no-explicit-any
	default: (props: any) => {
		// eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
		return <img {...props} priority="false" />;
	},
}));

jest.mock('@/helpers/pullBlogPost', () => ({
	getAllBlogPosts: jest.fn().mockReturnValue([
		{
			_id: '1',
			title: 'Post 1',
			createdAt: '2021-09-01T00:00:00.000Z',
			fullname: 'Author 1',
			othersizes: {
				tablet: 'image1.jpg',
			},
			postintro: '<p>Post 1 intro</p>',
			link: 'post-1',
			isLive: true,
		},
		{
			_id: '2',
			title: 'Post 2',
			createdAt: '2021-09-02T00:00:00.000Z',
			fullname: 'Author 2',
			othersizes: {
				tablet: 'image2.jpg',
			},
			postintro: '<p>Post 2 intro</p>',
			link: 'post-2',
			isLive: true,
		},
		{
			_id: '3',
			title: 'Post 3',
			createdAt: '2021-09-03T00:00:00.000Z',
			fullname: 'Author 3',
			othersizes: {
				tablet: 'image3.jpg',
			},
			postintro: '<p>Post 3 intro</p>',
			link: 'post-3',
			isLive: true,
		},
	]),
}));

describe('PostsList', () => {
	it('renders correctly with post data', async () => {
		const ComponentResolved = await PostsList();

		const MockComponent = () => ComponentResolved;
		const { container } = render(<MockComponent />);

		// Assert
		expect(container).toMatchSnapshot();
	});
});
