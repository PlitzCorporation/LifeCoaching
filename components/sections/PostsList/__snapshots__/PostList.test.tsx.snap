// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsList renders correctly with post data 1`] = `
<div>
  <div
    class="plitz-container py-10 lg:py-20"
  >
    <div
      class="grid grid-cols-1 lg:grid-cols-3 mb-20 last:mb-0 shadow-plitz-box-4-32"
    >
      <img
        alt="image3"
        class="object-cover min-h-full min-w-full max-h-full max-w-full border-t-10 border-plitz-primary"
        height="1000"
        priority="false"
        src="undefined/image3.jpg"
        width="1000"
      />
      <div
        class="relative lg:col-span-2"
      >
        <div
          class="hidden lg:block h-[10px] bg-plitz-primary"
        />
        <div
          class="h-full flex flex-col justify-center px-8 lg:px-20 py-8 lg:py-10"
        >
          <h3
            class="text-balance mb-3 lg:mb-5"
          >
            Post 3
          </h3>
          <small
            class="text-plitz-primary mb-7 lg:mb-10"
          >
            By 
            Author 3
             |
             
            September 02, 2021
          </small>
          <div
            class="text-plitz-body mb-7 lg:mb-10"
          >
            <p>
              Post 3 intro
            </p>
          </div>
          <div
            class="flex"
          >
            <a
              class="relative flex items-center bg-plitz-primary text-white px-5 py-3 hover:rounded-xl border border-plitz-primary md:px-7 lg:px-9 md:py-4 hover:border-plitz-accent hover:text-plitz-accent hover:bg-white transition-all hover:transition-all group shadow-2xl"
              data-testid="aTag"
              href="/blog/post-3"
            >
              <span
                class="translate-x-2 group-hover:-translate-x-1 transition-all"
              >
                Continue Reading
              </span>
              <svg
                class="fill-white translate-x-0 opacity-0 group-hover:fill-plitz-accent group-hover:opacity-100 group-hover:translate-x-1 transition-all"
                height="1em"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M440.6 273.4c4.7-4.5 7.4-10.8 7.4-17.4s-2.7-12.8-7.4-17.4l-176-168c-9.6-9.2-24.8-8.8-33.9.8s-8.8 24.8.8 33.9L364.1 232H24c-13.3 0-24 10.7-24 24s10.7 24 24 24h340.1L231.4 406.6c-9.6 9.2-9.9 24.3-.8 33.9s24.3 9.9 33.9.8l176-168z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-1 lg:grid-cols-3 mb-20 last:mb-0 shadow-plitz-box-4-32"
    >
      <img
        alt="image2"
        class="object-cover min-h-full min-w-full max-h-full max-w-full border-t-10 border-plitz-primary"
        height="1000"
        priority="false"
        src="undefined/image2.jpg"
        width="1000"
      />
      <div
        class="relative lg:col-span-2"
      >
        <div
          class="hidden lg:block h-[10px] bg-plitz-primary"
        />
        <div
          class="h-full flex flex-col justify-center px-8 lg:px-20 py-8 lg:py-10"
        >
          <h3
            class="text-balance mb-3 lg:mb-5"
          >
            Post 2
          </h3>
          <small
            class="text-plitz-primary mb-7 lg:mb-10"
          >
            By 
            Author 2
             |
             
            September 01, 2021
          </small>
          <div
            class="text-plitz-body mb-7 lg:mb-10"
          >
            <p>
              Post 2 intro
            </p>
          </div>
          <div
            class="flex"
          >
            <a
              class="relative flex items-center bg-plitz-primary text-white px-5 py-3 hover:rounded-xl border border-plitz-primary md:px-7 lg:px-9 md:py-4 hover:border-plitz-accent hover:text-plitz-accent hover:bg-white transition-all hover:transition-all group shadow-2xl"
              data-testid="aTag"
              href="/blog/post-2"
            >
              <span
                class="translate-x-2 group-hover:-translate-x-1 transition-all"
              >
                Continue Reading
              </span>
              <svg
                class="fill-white translate-x-0 opacity-0 group-hover:fill-plitz-accent group-hover:opacity-100 group-hover:translate-x-1 transition-all"
                height="1em"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M440.6 273.4c4.7-4.5 7.4-10.8 7.4-17.4s-2.7-12.8-7.4-17.4l-176-168c-9.6-9.2-24.8-8.8-33.9.8s-8.8 24.8.8 33.9L364.1 232H24c-13.3 0-24 10.7-24 24s10.7 24 24 24h340.1L231.4 406.6c-9.6 9.2-9.9 24.3-.8 33.9s24.3 9.9 33.9.8l176-168z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-1 lg:grid-cols-3 mb-20 last:mb-0 shadow-plitz-box-4-32"
    >
      <img
        alt="image1"
        class="object-cover min-h-full min-w-full max-h-full max-w-full border-t-10 border-plitz-primary"
        height="1000"
        priority="false"
        src="undefined/image1.jpg"
        width="1000"
      />
      <div
        class="relative lg:col-span-2"
      >
        <div
          class="hidden lg:block h-[10px] bg-plitz-primary"
        />
        <div
          class="h-full flex flex-col justify-center px-8 lg:px-20 py-8 lg:py-10"
        >
          <h3
            class="text-balance mb-3 lg:mb-5"
          >
            Post 1
          </h3>
          <small
            class="text-plitz-primary mb-7 lg:mb-10"
          >
            By 
            Author 1
             |
             
            August 31, 2021
          </small>
          <div
            class="text-plitz-body mb-7 lg:mb-10"
          >
            <p>
              Post 1 intro
            </p>
          </div>
          <div
            class="flex"
          >
            <a
              class="relative flex items-center bg-plitz-primary text-white px-5 py-3 hover:rounded-xl border border-plitz-primary md:px-7 lg:px-9 md:py-4 hover:border-plitz-accent hover:text-plitz-accent hover:bg-white transition-all hover:transition-all group shadow-2xl"
              data-testid="aTag"
              href="/blog/post-1"
            >
              <span
                class="translate-x-2 group-hover:-translate-x-1 transition-all"
              >
                Continue Reading
              </span>
              <svg
                class="fill-white translate-x-0 opacity-0 group-hover:fill-plitz-accent group-hover:opacity-100 group-hover:translate-x-1 transition-all"
                height="1em"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M440.6 273.4c4.7-4.5 7.4-10.8 7.4-17.4s-2.7-12.8-7.4-17.4l-176-168c-9.6-9.2-24.8-8.8-33.9.8s-8.8 24.8.8 33.9L364.1 232H24c-13.3 0-24 10.7-24 24s10.7 24 24 24h340.1L231.4 406.6c-9.6 9.2-9.9 24.3-.8 33.9s24.3 9.9 33.9.8l176-168z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
